PythonVisualizations/BinaryTreeBase.py

```diff
@@ -0,0 +1,515 @@
+
+      self.moveItemsTo(arrow, (toPos,), steps = numSteps)
+
+   # calculate the coordinates for the node shape
+   def calculateCoordinates(self, parent, level, childDirection):
+        if level == 0:
+            return self.ROOT_X0, self.ROOT_Y0
+        elif childDirection == Child.LEFT:
+            return parent.coords[0] - 1/2**level* (self.NODE_X_GAP-self.CIRCLE_SIZE), self.ROOT_Y0+level* self.NODE_Y_GAP
+        else:
+            return parent.coords[0] + 1/2**level* (self.NODE_X_GAP-self.CIRCLE_SIZE), self.ROOT_Y0+level* self.NODE_Y_GAP
+
+   # calculate the coordinates for the line attached to the node
+   def calculateLineCoordinates(self, node):
+      x1 = node.coords[0]
+      y1 = node.coords[1] +self.CIRCLE_SIZE
+
+      # where would the child be drawn?
+      level = self.getLevel(node)+1
+      x2, y2 = self.calculateCoordinates(node, level, Child.LEFT)
+      x3, y3 = self.calculateCoordinates(node, level, Child.RIGHT)
+      y2 -= self.CIRCLE_SIZE
+      y3 -= self.CIRCLE_SIZE
+
+      return [(x1, y1, x2, y2), (x1, y1, x3, y3)]
+
+   # highlight the left line of the node
+   def highlightLeftLine(self, node, color = "red"):
+      # get all the items with the node's tag
+      nodeFamily = set(self.canvas.find_withtag(node.tag))
+      # get all the lines that are left
+      lines = set(self.canvas.find_withtag("left line"))
+
+      line = (lines & nodeFamily).pop()
+
+      self.canvas.itemconfig(line, fill=color)
+
+   # unhighlight the left line of the node
+   def unHighlightLeftLine(self, node):
+      # get all the items with the node's tag
+      nodeFamily = set(self.canvas.find_withtag(node.tag))
+      # get all the lines that are left
+      lines = set(self.canvas.find_withtag("left line"))
+
+      line = (lines & nodeFamily).pop()
+
+      self.canvas.itemconfig(line, fill="black")
+
+   # highlight the right line of the node
+   def highlightRightLine(self, node, color="red"):
+      # get all the items with the node's tag
+      nodeFamily = set(self.canvas.find_withtag(node.tag))
+      # get all the lines that are left
+      lines = set(self.canvas.find_withtag("right line"))
+
+      line = (lines & nodeFamily).pop()
+
+      self.canvas.itemconfig(line, fill= color)
+
+   # unhighlight the right line of the node
+   def unHighlightRightLine(self, node):
+      # get all the items with the node's tag
+      nodeFamily = set(self.canvas.find_withtag(node.tag))
+      # get all the lines that are left
+      lines = set(self.canvas.find_withtag("right line"))
+
+      line = (lines & nodeFamily).pop()
+
+      self.canvas.itemconfig(line, fill="black")
+
+   # highlight the circle associated with the node
+   def highlightCircle(self, node, color="red"):
+      self.canvas.itemconfig(node.drawable.display_shape, outline=color)
+
+   # unhighlight the circle associated with the node
+   def unHighlightCircle(self, node):
+      self.canvas.itemconfig(node.drawable.display_shape, outline="black")
+
```

Comment:
Similar to the lines, I think we need more than color changes here to highlight the circles on nodes.  Making the border thicker would probably be enough.

Note Tk supports different appearances for objects based on their state.  There is a set of parameters for `active` state and `disabled` state.  It might make sense to use those and only use `itemconfigure` to change the item's `state`.  That state, is also tied to other things like keyboard focus and selectability.

Reply:
Changes implemented in 9937d21